packages/salesforce/src/Adaptor.js

```diff
@@ -659,56 +659,103 @@ function setApiVersion(apiVersion) {
 
   return version;
 }
+
 /**
- * Creates a connection.
+ * Establish a connection to Salesforce using username and password.
  * @example
- * createConnection(state)
+ * createBasicAuthConnection(state)
  * @function
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
+async function createBasicAuthConnection(state) {
+  console.log('Attempting to connect with Basic Auth');
+
+  const { loginUrl, apiVersion, username, password, securityToken } =
+    state.configuration;
 
-  if (!loginUrl) {
-    throw new Error('loginUrl missing from configuration.');
+  if (!loginUrl || !username || !password) {
+    throw new Error(
+      'Incomplete configuration for basic authentication. Provide loginUrl, username, password, and securityToken if required.'
+    );
   }
 
+  const connectionOptions = { loginUrl };
+
+  if (apiVersion) {
+    connectionOptions.version = setApiVersion(apiVersion);
```

Comment:
hmm. `setApiVersion` doesn't actually set anything.

I'm not sure how big the change is, but can we consider either:
a) changing this to `getApiVersion(state)`
b)  changing it to `setApiVersion(state, connectionOptions)` but pass in the `connectionOptions` object. The function will set the `apiVersion` key if it exists on state.config/

Reply:
@josephjclark I choose option `b`. But i think i need to refactor my implementation of `setApiVersion`. Because i don't want to set the api version if i haven't specified the version in `state.configuration`. and currently my implementation of `setApiVersion` defaults to `v52.0`

Reply:
If it's too complicated you can leave it - it's just something that caught my eye.

Reply:
It's not a big change, i have pushed the change with `setApiVersino(state, connOpts)` @josephjclark